src/hotspot/share/memory/universe.cpp

```diff
@@ -236,29 +234,38 @@ void Universe::metaspace_pointers_do(MetaspaceClosure* it) {
   _do_stack_walk_cache->metaspace_pointers_do(it);
 }
 
+#if INCLUDE_CDS_JAVA_HEAP
+void Universe::set_archived_mirror_index(BasicType t, int index) {
```

Comment:
Is it called during dump time only?
If so, maybe add an assert?
`assert(DumpSharedSpaces, "dump-time only");`

Reply:
Fixed as suggested.